### PR TITLE
Unify akka logs with neo logs

### PR DESCRIPTION
**Pull request by shargon**
Required for fix: https://github.com/neo-project/neo/issues/1245

### CODE REVIEW

File: src/neo/Utility.cs [hunk 1]
```diff
@@ -6,6 +8,32 @@ namespace Neo
...
+
+            private void OnLog(LogLevel level, string message)
+            {
+                Log<NeoSystem>(level, message);
+            }
+        }
+
+        public delegate void delOnLog(string source, LogLevel level, string message);
+
+        public static event delOnLog OnLog;
```
Comment (erikzhang) [inline src/neo/Utility.cs:35]: Is this event used in anywhere?

Comment (shargon) [reply src/neo/Utility.cs:35]: It could be used for merge the errors with the `error. log` file in neo-cli

Comment (erikzhang) [reply src/neo/Utility.cs:35]: You can register cli as a plugin. Or just create a plugin in cli.

File: src/neo/LogLevel.cs [hunk 2]
```diff
@@ -2,10 +2,10 @@ namespace Neo
...
-        Fatal,
-        Error,
-        Warning,
-        Info,
-        Debug
+        Debug = 0,
+        Info = 1,
+        Warning = 2,
+        Error = 3,
+        Fatal = 4
```
Comment (erikzhang) [inline src/neo/LogLevel.cs:9]: Why modify the order?

File: src/neo/Utility.cs [hunk 3]
```diff
@@ -25,5 +49,10 @@ public static void Log(string source, LogLevel level, string message)
             foreach (ILogPlugin plugin in Plugin.Loggers)
                 plugin.Log(source, level, message);
         }
+
+        public static void Log<T>(LogLevel level, string message)
+        {
+            Log(typeof(T).Name, level, message);
+        }
```
Comment (erikzhang) [inline src/neo/Utility.cs:56]: I think this overload is unnecessary.

File: src/neo/LogLevel.cs (hunk 2 above)
Comment (shargon) [reply src/neo/LogLevel.cs:9]: Because it's easier if we want to log `Error` and `Fatal`, we can use > `Error`

File: src/neo/Plugins/Plugin.cs [hunk 4]
```diff
@@ -109,7 +109,7 @@ private static Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEven
             }
             catch (Exception ex)
             {
-                Utility.Log(nameof(Plugin), LogLevel.Error, $"Failed to resolve assembly or its dependency: {ex.Message}");
+                Utility.Log(an.Name, LogLevel.Error, $"Failed to resolve assembly or its dependency: {ex.Message}");
```
Comment (erikzhang) [inline src/neo/Plugins/Plugin.cs:112]: The `source` is similar to the category, so I think it is better to use `nameof(Plugin)`.

File: src/neo/Plugins/Plugin.cs [hunk 5]
```diff
@@ -137,7 +137,7 @@ private static void LoadPlugin(Assembly assembly)
                 }
                 catch (Exception ex)
                 {
-                    Utility.Log(nameof(Plugin), LogLevel.Error, $"Failed to initialize plugin: {ex.Message}");
+                    Utility.Log(type.Name, LogLevel.Error, $"Failed to initialize plugin: {ex.Message}");
```
Comment (erikzhang) [inline src/neo/Plugins/Plugin.cs:140]: Same here.

File: src/neo/Utility.cs [hunk 6]
```diff
@@ -6,6 +8,28 @@ namespace Neo
...
     {
+        internal class Logger : ReceiveActor
+        {
+            public Logger()
+            {
+                Receive<Debug>(e => OnLog(LogLevel.Debug, e.ToString()));
+                Receive<Info>(e => OnLog(LogLevel.Info, e.ToString()));
+                Receive<Warning>(e => OnLog(LogLevel.Warning, e.ToString()));
+                Receive<Error>(e => OnLog(LogLevel.Error, e.ToString()));
+                Receive<InitializeLogger>(_ => Init(Sender));
```
Comment (erikzhang) [inline src/neo/Utility.cs:19]: No `Fatal`?

Comment (shargon) [reply src/neo/Utility.cs:19]: Akka can't find `Fatal`, I think that akka only have error, warning, info, and debug

**Pull request merged**